src/probability/conditional.lean

```diff
@@ -0,0 +1,129 @@
+/-
+Copyright (c) 2021 Rishikesh Vaishnav. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rishikesh Vaishnav
+-/
+
+import probability.independence
+
+/-!
+# Conditional Probablity
+
+This file defines conditional probability and includes basic results relating to it.
+
+Given some measure `μ` defined on a measure space on some type `α` and some `s : set α`,
+we define the measure of `μ` conditioned on `s` as the restricted measure scaled by
+the inverse of the measure of `s`: `(μ s)⁻¹ • μ.restrict s`. The scaling
+ensures that this is a probability measure (when `μ` is a finite measure).
+
+From this definition, we derive the "axiomatic" definition of conditional probability
+based on application: for any `s t : set α`, we have `μ[t|s] = (μ s)⁻¹ * μ (s ∩ t)`.
+
+## Main Statements
+
+* `cond_cond_eq_cond_inter`: conditioning on one set and then another is equivalent
+  to conditioning on their intersection.
+* `bayes`: Bayes' Theorem.
+
+## Notations
+
+This file uses the local notation `μ[|s]` the measure of `μ` conditioned on `s`,
+and `μ[t|s]` for the probability of `t` given `s` under `μ` (equivalent to the
+application `μ[|s] t`.
+
+## Implementation notes
+
+Because we have the alternative measure restriction application principles
+`measure.restrict_apply` and `measure.restrict_apply'`, which require 
+measurability of the restricted and restricting sets, respectively,
+many of the theorems here will have corresponding alternatives as well.
+For the sake of brevity, we've chosen to only go with `measure.restrict_apply'`
+for now, but the alternative theorems can be added if needed.
+-/
+
+noncomputable theory
+
+open measure_theory measurable_space
+
+variables {α : Type*} [m : measurable_space α] (μ : measure α)
+
+namespace probability_theory
+
+section definitions
+
+/-- The conditional probability measure of measure `μ` on set `s` is `μ` restricted to `s` 
+and scaled by the inverse of `μ s` (to make it a probability measure):
+`(μ s)⁻¹ • μ.restrict s`. -/
+def cond (s : set α) : measure α :=
+  (μ s)⁻¹ • μ.restrict s
+
+end definitions
+
+local notation  μ `[` s `|` t `]` := cond μ t s
+local notation  μ `[|` t`]` := cond μ t
```

Comment:
Noticed that this conflicts with some existing notation for conditional expectation -- let me know what you think is best here, will be happy to change.